test/src/tests-serverModel.ts

```diff
@@ -0,0 +1,1676 @@
+                eventType: Types.EventType.Other,
+            };
+
+            const actualEventObject = mParticle
+                .getInstance()
+                ._ServerModel.createEventObject(event) as IUploadObject;
+
+            expect(
+                actualEventObject.SourceMessageId,
+                'SourceMessageId'
+            ).to.not.equal(null);
+        });
+
+        it('returns null if _Store does not have a sessionId', () => {
+            const mPStore = mParticle.getInstance()._Store;
+            mPStore.sessionId = null;
+
+            const event: BaseEvent = {
+                name: 'Test Opt Out Event',
+                messageType: Types.MessageType.CustomEvent,
+            };
+
+            const actualEventObject = mParticle
+                .getInstance()
+                ._ServerModel.createEventObject(event) as IUploadObject;
+
+            expect(actualEventObject).to.equal(null);
+        });
+
+        it('returns null if _Store has Webview Bridge Disabled', () => {
+            const mPStore = mParticle.getInstance()._Store;
+
+            mPStore.sessionId = null;
+            mPStore.webviewBridgeEnabled = false;
+
+            const event: BaseEvent = {
+                name: 'Test Opt Out Event',
+                messageType: Types.MessageType.CustomEvent,
+            };
+
+            const actualEventObject = mParticle
+                .getInstance()
+                ._ServerModel.createEventObject(event) as IUploadObject;
+
+            expect(actualEventObject).to.equal(null);
+        });
+
+        it('returns null if event is invalid', () => {
+            const event: BaseEvent = ({} as unknown) as BaseEvent;
+
+            expect(ServerModel.createEventObject(event)).to.eql(null);
+        });
+
+        it('should support toEventAPIObject', () => {
+            const eventAPIObject = {
+                EventName: 'Test Event Object Override',
+                EventCategory: Types.EventType.Media,
+                EventDataType: Types.MessageType.Media,
+            };
+
+            const event = {
+                name: 'Test Event Object',
+                messageType: Types.MessageType.PageEvent,
+                eventType: Types.EventType.Other,
+                toEventAPIObject: () => eventAPIObject,
+            } as BaseEvent;
+
+            const actualEventObject = mParticle
+                .getInstance()
+                ._ServerModel.createEventObject(event);
+
+            expect(actualEventObject.EventName, 'EventName').to.equal(
+                'Test Event Object Override'
+            );
+            expect(actualEventObject.EventCategory, 'EventCategory').to.equal(
+                Types.EventType.Media
+            );
+            expect(actualEventObject.EventDataType, 'EventDataType').to.equal(
+                Types.MessageType.Media
+            );
+        });
+    });
+
+    describe('#convertEventToDTO', () => {
+        it('should return defaults', () => {
```

Comment:
Agreed. Does this make more sense: 

```suggestion
        it('should convert an event to a DTO', () => {
```